jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/ImageParameters.java

```diff
@@ -70,22 +71,42 @@ public void setCredHelper(String credHelper) {
 
   @Nested
   @Optional
-  AuthConfiguration getAuth() {
+  AuthParameters getAuth() {
     return auth;
   }
 
-  public void auth(Action<? super AuthConfiguration> action) {
+  public void auth(Action<? super AuthParameters> action) {
     action.execute(auth);
   }
 
-  /** Converts the {@link ImageParameters} to an {@link Authorization}. */
+  /**
+   * Converts the {@link ImageParameters} to an {@link Authorization}.
+   *
+   * @param logger the {@link JibLogger} used to print warnings
+   * @param imageProperty the image configuration's name (i.e. "from" or "to")
+   * @return the {@link Authorization}, or null if the username and password aren't both configured
+   */
   @Internal
   @Nullable
-  Authorization getImageAuthorization() {
+  Authorization getImageAuthorization(JibLogger logger, String imageProperty) {
     if (auth.getUsername() == null || auth.getPassword() == null) {
+      if (auth.getPassword() != null) {
```

Comment:
Wouldn't this warn if `jib.auth` was null too? (aka should not warn if user did not define `jib.auth`)

Reply:
If the user doesn't configure auth, both username and password should both be null.

Reply:
Ah right - maybe it would make more sense to split this logic into 3 parts (vs entwining the 3 cases in if-else-fallthrough):

both are null - return null
only password is null - warn about password
only username is null - warn about username